components/mpas-ocean/src/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -301,10 +350,11 @@ subroutine ocn_tracer_hmix_Redi_tend(meshPool, layerThickness, layerThickEdgeMea
                                 invAreaCell
                   redi_term2_edge(iTr, k, iEdge) = -flux_term2
 
-                  dTracerDx = (tracers(iTr, k, cell2) - tracers(iTr, k, cell1))*dvEdge(iEdge)*0.25_RKIND
-                  fluxRediZTop(iTr, k) = fluxRediZTop(iTr, k) + &
-                                         0.5_RKIND*(slopeTriadUp(k, iCellSelf, iEdge) + slopeTriadDown(k - 1, iCellSelf, iEdge)) &
-                                         *dTracerDx
+                  ! term 3 : d/dz div( S grad psi )
+                  ! includes weights for edge-to-cell remap
+                  fluxRediZTop(iTr, k) = fluxRediZTop(iTr, k) + 0.125_RKIND*dvEdge(iEdge)*( &
+                     slopeTriadDown(k - 1, iCellSelf, iEdge)*(tracers(iTr, k - 1, cell2) - tracers(iTr, k - 1, cell1)) + &
+                     slopeTriadUp(k, iCellSelf, iEdge)*(tracers(iTr, k, cell2) - tracers(iTr, k, cell1)))
```

Comment:
@dengwirda I think your altered version here is correct. From [Griffies 1998](https://journals.ametsoc.org/view/journals/phoc/28/5/1520-0485_1998_028_0805_idiazc_2.0.co_2.xml):
![image](https://github.com/E3SM-Ocean-Discussion/E3SM/assets/4036816/504cdf6f-0631-4436-a0d0-9e3f1bf61af2)
We are looking at $S \cdot \nabla \psi$, which are the red and green boxes in this sum. See that the green box has a vertical index $k+kr$. The $ip$ and $kr$ both index from 0 to 1 to sum over the four directions of triads, where kr is the vertical index. That shows that the tracer should be taken from the same vertical level as the corner of the triad, which is what you have done. Thanks!  I'll try both versions for my convergence tests and see if they differ.